File: code/IceCream.java
import java.util.*;
public class IceCream{
    static int n,m;
    static int[][] graph = new int[1000][1000];
    static boolean dfs(int a, int b){
        //주어진 범위를 벗어나는 경우
        if (a<0 || a>n-1 || b<0 || b>n-1 ) return false;
        if(graph[a][b] ==0){
            graph[a][b] =1;
            dfs(a, b-1);
            dfs(a, b-1);
            dfs(a+1, b);
            dfs(a, b+1);
            return true;
        }
        return false;
    }
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        n =sc.nextInt();
        m = sc.nextInt();
        for (int i=0;i<n;i++){
            String str = sc.nextLine();
            for(int j=0;j<m;j++){
                graph[i][j] = str.charAt(i)-'0';
            }
        }
        int result=0;
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if (dfs(i,j)) result+=1;
            }
        }
        System.out.println(result);

    }

}

File: code/ice_cream.py
def dfs(x,y):
    if (x<0 or x>n or y<0 or y>n): return False
    if graph[x][y] ==0:
        graph[x][y] =1
        dfs(x-1,y)
        dfs(x+1,y)
        dfs(x,y+1)
        dfs(x,y-1)

n,m = 4,5
graph=[]
for i in range(n):
    graph.apppend(list(map,input()))
result = 0
for i in range(n):
    for j in range(m):
        if dfs(i,j) ==True: result+=1
print(result)        
